### PR TITLE
media-plugins/vdr-devstatus: fix for >vdr-2.3

### PR DESCRIPTION
**Pull request by madmartin**
add misc patches to make the plugin work together
with >media-video/vdr-2.3

Package-Manager: Portage-2.3.78, Repoman-2.3.17
Signed-off-by: Martin Dummer <martin.dummer@gmx.net>

### CODE REVIEW

**Comment by madmartin** [comment]
@juippis can you please review here, too?

File: media-plugins/vdr-devstatus/metadata.xml [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
+		<maintainer type="project">
+			<email>proxy-maint@gentoo.org</email>
+			<name>Proxy Maintainers</name>
+		</maintainer>
+		<maintainer type="project">
+			<email>vdr@gentoo.org</email>
+			<name>Gentoo VDR Project</name>
+		</maintainer>
+		<longdescription>
+			VDR Plugin to show the current status of dvb devices (tuned channes, recordings) on OSD
```
Comment (juippis) [inline media-plugins/vdr-devstatus/metadata.xml]: This isn't any longer than the default ebuild description and could be removed :)

File: media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+DEPEND=">=media-video/vdr-1.6.0"
+RDEPEND="${DEPEND}"
+
+QA_FLAGS_IGNORED="
+	usr/lib/vdr/plugins/libvdr-.*
+	usr/lib64/vdr/plugins/libvdr-.*"
+
+src_prepare() {
+	# remove untranslated .po files
+	rm "${S}"/po/{ca_ES,cs_CZ,da_DK,el_GR,et_EE,hr_HR,hu_HU,nl_NL,nn_NO,pl_PL,pt_PT,ro_RO,sl_SI,sv_SE,tr_TR}.po \
```
Comment (juippis) [inline media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild]: Why is this necessary though? Do they get installed otherwise?

File: media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+inherit vdr-plugin-2
+
+DESCRIPTION="VDR plugin: display the usage status of the available DVB devices"
+HOMEPAGE="http://www.u32.de/vdr.html"
+SRC_URI="http://www.u32.de/download/${P}.tgz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~x86"
```
Comment (juippis) [inline media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild]: You can't add these keywords, as you're not part of the dedicated arch teams. You need to ask keywording via Bugzilla, let me know if you need help doing that.

File: media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+DEPEND=">=media-video/vdr-1.6.0"
```
Comment (juippis) [inline media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild]: Always satisfied etc :)

File: media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+src_prepare() {
+	# remove untranslated .po files
+	rm "${S}"/po/{ca_ES,cs_CZ,da_DK,el_GR,et_EE,hr_HR,hu_HU,nl_NL,nn_NO,pl_PL,pt_PT,ro_RO,sl_SI,sv_SE,tr_TR}.po \
+		|| die "failed to remove untranslated .po files"
+
+	vdr-plugin-2_src_prepare
+
+	# some patches from debian/e-tobi
+	eapply "${FILESDIR}/recordingfound.patch"
```
Comment (juippis) [inline media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild]: I see you added these patches. Remember the naming "rules" we try to follow, 
https://dev.gentoo.org/~vapier/clean-patches
Ie add `${P}` to patch name.

File: media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	eapply "${FILESDIR}/channelswitch.patch"
+	eapply "${FILESDIR}/dvb-adapter-frontend.patch"
+	eapply "${FILESDIR}/memoryleak.patch"
+
+	if has_version ">=media-video/vdr-2.3"; then
+		eapply "${FILESDIR}/devstatus-vdr2.3.patch"
```
Comment (juippis) [inline media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild]: Can the patch be written so it doesn't break backwards-compatibility? Then it could always be applied, and you could use a cleaner PATCHES array method for all of them.

File: media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild (hunk 2 above)
Comment (madmartin) [reply media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild]: yes, they would be installed.

File: media-plugins/vdr-devstatus/metadata.xml (hunk 1 above)
Comment (madmartin) [reply media-plugins/vdr-devstatus/metadata.xml]: ok. not on my "git blame" report...

File: media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild (hunk 3 above)
Comment (madmartin) [reply media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild]: learned that already... arch keywords have to be added through bugzilla...

File: media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild (hunk 4 above)
Comment (madmartin) [reply media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild]: got it!

File: media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild (hunk 5 above)
Comment (madmartin) [reply media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild]: Have read that page and renamed patches accordingly.

File: media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild (hunk 6 above)
Comment (madmartin) [reply media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild]: yes, I refactored the last patch and tested compilation against vdr-2.2.0 and vdr-2.4.1

File: media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+QA_FLAGS_IGNORED="
+	usr/lib/vdr/plugins/libvdr-.*
+	usr/lib64/vdr/plugins/libvdr-.*"
+
+src_prepare() {
+	# remove untranslated .po files
+	rm "${S}"/po/{ca_ES,cs_CZ,da_DK,el_GR,et_EE,hr_HR,hu_HU,nl_NL,nn_NO,pl_PL,pt_PT,ro_RO,sl_SI,sv_SE,tr_TR}.po \
+		|| die "failed to remove untranslated .po files"
+
+	local PATCHES=(
```
Comment (juippis) [inline media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild]: Move this somewhere above src_prepare to make it proper global var, and then I'm happy :) 
(just see examples from about any ebuild using PATCHES array)

Comment (madmartin) [reply media-plugins/vdr-devstatus/vdr-devstatus-0.4.1-r1.ebuild]: its done.
